api/web3server.go

```diff
@@ -433,7 +435,24 @@ func (svr *web3Handler) sendRawTransaction(in *gjson.Result) (interface{}, error
 		return nil, errInvalidFormat
 	}
 	// parse raw data string from json request
-	tx, sig, pubkey, err := action.DecodeRawTx(dataStr.String(), svr.coreService.EVMNetworkID())
+	var (
+		tx       *types.Transaction
+		sig      []byte
+		pubkey   crypto.PublicKey
+		err      error
+		encoding iotextypes.Encoding
+		cs       = svr.coreService
+	)
+	if g := cs.Genesis(); g.IsSumatra(cs.TipHeight()) {
+		tx, encoding, sig, pubkey, err = action.DecodeEtherTx(dataStr.String())
+		if !tx.Protected() && tx.ChainId().Uint64() != uint64(cs.EVMNetworkID()) {
```

Comment:
If `err != nil`, what will happen here? tx may not be initialized properly, and we shouldn't have any hidden assumption of the default value of tx

Reply:
good catch, fixed

Reply:
common `err` is checked at L456 below, but you are right, logically should check `err` first

Reply:
your modification here answers your own question:
```just curious, why want to return nil for tx specifically? we are returning the default value of these types here, and there's err to be checked```
so, let's return `nil` inside of that function, and also split it into two functions pls